### PR TITLE
Supervisor start course

### PR DESCRIPTION
**Pull request by lethihaivan**
Screenshot from 2021-04-08 11-41-03
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/34983)

## WHAT
As a supervisor, I can start a course

## Evidence (Screenshot or Video)
Screenshot from 2021-04-08 11-46-38

### CODE REVIEW

**Comment by lethihaivan** [comment]
ready

File: app/controllers/supervisor/courses_controller.rb [hunk 1]
```diff
@@ -39,6 +39,22 @@ def update
     end
   end
 
+  def start
+    ActiveRecord::Base.transaction do
+      cou_sub_ids = @course.course_subjects
+                           .pluck(:id)
+                           .map{|id| {course_subject_id: id, status: :joined}}
+      @course.update! status: :start
+      @course.user_courses.update_all active: true
```
Comment (vanvtt-0952) [inline app/controllers/supervisor/courses_controller.rb]: `update_all` co' rollback data dc ko em

File: app/controllers/supervisor/courses_controller.rb [hunk 2]
```diff
@@ -39,6 +39,22 @@ def update
     end
   end
 
+  def start
+    ActiveRecord::Base.transaction do
+      cou_sub_ids = @course.course_subjects
+                           .pluck(:id)
+                           .map{|id| {course_subject_id: id, status: :joined}}
+      @course.update! status: :start
```
Comment (vanvtt-0952) [inline app/controllers/supervisor/courses_controller.rb]: Tim- hieu ve enum, enum co' ho tro ne-

File: app/controllers/supervisor/courses_controller.rb (hunk 1 above)
Comment (lethihaivan) [reply app/controllers/supervisor/courses_controller.rb:48]: Screenshot from 2021-04-08 15-40-47
em thấy vẫn rollback đc chị

**Comment by lethihaivan** [comment]
ready

**Comment by vanvtt-0952** [comment]
commented
comment trucwj tieesp

**Comment by lethihaivan** [comment]
ready

**Pull request merged**